packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -154,13 +178,25 @@ class CachedArtifacts extends Artifacts {
   }
 
   String _getIosArtifactPath(Artifact artifact, TargetPlatform platform, BuildMode mode) {
-    final String engineDir = _getEngineArtifactsPath(platform, mode);
     switch (artifact) {
       case Artifact.genSnapshot:
       case Artifact.snapshotDart:
       case Artifact.flutterFramework:
       case Artifact.frontendServerSnapshotForEngineDartSdk:
+        final String engineDir = _getEngineArtifactsPath(platform, mode);
         return fs.path.join(engineDir, _artifactToFileName(artifact));
+      case Artifact.ideviceId:
+      case Artifact.ideviceinfo:
+      case Artifact.idevicescreenshot:
+      case Artifact.idevicesyslog:
+      case Artifact.idevicename:
+        return fs.path.join(cache.getArtifactDirectory('libimobiledevice').path, _artifactToFileName(artifact));
```

Comment:
nit: Instead of repeating `_artifactToFileName(artifact)`, how about assigning to a variable above the switch.

Reply:
done